File: gdbm/python/update/gdbm_update.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	gdbm_update.py
#
#					Jul/30/2014
import	sys
import	string
import	dbm
#
sys.path.append ('/var/www/data_base/common/python_common')
#
from dbm_manipulate import dbm_update_proc
# -------------------------------------------------------------
print ("*** 開始 ***")
#
key_in = sys.argv[1]
population_in = int (sys.argv[2])
print ("%s\t%d" % (key_in, population_in))
#
db_name = "/var/tmp/gdbm/cities.pag";
dd = dbm.open (db_name,"c")
dbm_update_proc	(dd,key_in,population_in)
#
dd.close ()
#
print ("*** 終了 ***")
# -------------------------------------------------------------

File: gdbm/php/update/gdbm_update.php
#! /usr/bin/php
<?php
// ------------------------------------------------------------------
//	gdbm_update.php
//
//					Jun/11/2015
//
// ------------------------------------------------------------------
$path="/var/www/data_base/common/php_common";
set_include_path (get_include_path() . PATH_SEPARATOR . $path);

include "dba_manipulate.php";

// ------------------------------------------------------------------
print "*** 開始 ***\n";
$file_name=$argv[1];
$key_in = $argv[2];
$population_in = $argv[3];

$dbh = dba_open($file_name,"c","gdbm");

$json_string = dba_fetch($key_in, $dbh);
echo $json_string . "\n";

$city = json_decode ($json_string);
echo $key_in . "\t";
echo $city->name . "\t";
echo $city->population . "\t";
echo $city->date_mod;
echo "\n";

date_default_timezone_set('Asia/Tokyo');
$today = date ("Y-m-d");

$encode = value_to_json_proc ($city->name,$population_in,$today);

dba_replace ($key_in,$encode,$dbh);

dba_close ($dbh);

print "*** 終了 ***\n";
// ------------------------------------------------------------------
?> 
